src/audio/src/src.c

```diff
@@ -391,11 +390,11 @@ static void src_2s(struct comp_dev *dev, const struct audio_stream __sparse_cach
 	 */
 	if (s1_blk_out > sbuf_free) {
 		s1.times = sbuf_free / (cd->src.stage1->blk_out * nch);
-		s1_blk_in = s1.times * cd->src.stage1->blk_in * nch;
 		s1_blk_out = s1.times * cd->src.stage1->blk_out * nch;
 		comp_dbg(dev, "s1.times = %d", s1.times);
 	}
 
+	s1_blk_in = s1.times * cd->src.stage1->blk_in * nch;
```

Comment:
no need add, just delete line 394 is fine, and keep line 385, minimum change, right?

Reply:
I hope I understood - the s1.times is re-calculated if s1_blk_out exceeds the intermediate buffer (between s1 and s2)  free. I don't think it can be simplified. I need to make a first shot and adjust it to buffers avail and free. It would be quite similar the other way to calculate.

Reply:
yes, you are right, Seppo. s1.times is changing, so this change is fine.